### ISSUE TITLE
Workflow: Git and createParameterSet

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/picongpu#501]
### Document a work flow to keep simulation input in an easy upgradeable way.

A good workflow to run and maintain a simulation setup with git is like that:
- checkout a _fresh branch_ from the latest `master` (or `dev`)
- choose the example that is the _closest_ that fits your needs and _modify it_
- commit your changes

Now run createParameterSets and compile, run, ...

Examples and local simulations become hard to maintain/update when one checks in the output of `createParameterSets`.

Also: the new files from that command are no game changer since we adapted the strategy to always use a _minimal default PIC cycle_ in the file in `src/picongpu/include/simulation_defines/`.
### Other Stuff

We implemented that one can directly compile such a "reduced example" (that was not filled up with files from `createParameterSet`).
Questions:
- [ ] does that still work?
- [ ] could one in principle skip `createParameterSet` completely from the workflow? In that case the question remains how to fill up the other files necessary to run a successful `tbg` command on it.

**Comment by BrianMarre** [comment]
@PrometheusPi is this issue still relevant?

**Comment by PrometheusPi** [comment]
It's partially described in here: https://picongpu.readthedocs.io/en/latest/usage/basics.html
But not in it the full context that@ax3l envisioned. 
@ComputationalRadiationPhysics/picongpu-developers Do you think a more detailed documentation is needed?

**Comment by chillenzer** [comment]
From skim-reading through the linked doc page, I'd say what's missing there (as compared to the described workflow here) is version-controlling your changes. That sounds indeed like a very good idea, capable of leveraging the full `diff`/`rebase`/... capabilities of `git` for updating everything. I think integrating that would be quite doable and worthwhile. One could either sprinkle in a few sentences here and there mentioning `git` or provide a subsection right before [Details on the commands above](https://picongpu.readthedocs.io/en/latest/usage/basics.html#details-on-the-commands-above) summarising the changes (or rather additions) to the described workflow to incorporate a version-controlled setup. Personally, I think I've got a preference which way to go but feel free to discuss.

**Comment by PrometheusPi** [comment]
I agree with @chillenzer that this would be worth documenting, ideally by someone with no thesis to write and experience in developing setups. @steindev and I should play tick-tack-toe. Whoever loses has to write the chapter. 🍀 (@psychocoderHPC you are safe, your task list is too long.)

**Comment by chillenzer** [comment]
Then, please, go ahead. I've prepared a board for you:
```
-------------
| x |   |   |
-------------
|   |   |   |
-------------
|   |   |   |
-------------
```
`x`: @steindev
`o`: @PrometheusPi

**Comment by steindev** [comment]
👆 @PrometheusPi

**Comment by berceanu** [comment]
What if it's a draw, as it's normally the case with perfect play from both sides?

**Comment by ax3l** [comment]
```
-------------
| x |   |   |
-------------
|   |   |   |
-------------
|   |   | v |
-------------
```
`x`: @steindev
`o`: @PrometheusPi
`v`: @ax3l

**Comment by PrometheusPi** [comment]
```
-------------
| x |   | o |
-------------
|   | o |   |
-------------
| o |   | v |
-------------
```

😜 - who said we need to follow rules? 

PS: so @chillenzer has to write the chapter according to the rules ;P

**Comment by chillenzer** [comment]
From a wise, old man in a distant past I learnt the following lesson I've kept in my heart for ever since:

> 😜 - who said we need to follow rules?

**Comment by chillenzer** [comment]
Now, I'm wondering if there's an option to "lock" an issue, so @steindev cannot unassign himself. XD
